pkg/models/sigv4.go

```diff
@@ -90,9 +89,28 @@ func (m *SigV4Middleware) signRequest(req *http.Request) (http.Header, error) {
 }
 
 func (m *SigV4Middleware) signer() (*v4.Signer, error) {
-	c, err := m.credentials()
-	if err != nil {
-		return nil, err
+	authType := AuthType(m.Config.AuthType)
+
+	var c *credentials.Credentials
+	switch authType {
+	case Default:
+		c = nil
+	case Keys:
+		c = credentials.NewStaticCredentials(m.Config.AccessKey, m.Config.SecretKey, "")
+	case Credentials:
+		c = credentials.NewSharedCredentials("", m.Config.Profile)
+	}
+
+	// passing nil credentials will force AWS to allow more complete credential chain vs the explicit default
+	if c == nil {
+		s, err := session.NewSession(&aws.Config{
+			Region:      aws.String(m.Config.Region),
+			Credentials: nil},
```

Comment:
There's no need to provide the default value.
```suggestion
		},
```